crates/solver/src/settlement_post_processing/mod.rs

```diff
@@ -113,13 +129,27 @@ impl PostProcessing for PostProcessingPipeline {
         .await;
 
         // an error will leave the settlement unmodified
-        optimize_unwrapping(
+        let optimized_solution = optimize_unwrapping(
             optimized_solution,
             &simulator,
             &self.buffer_retriever,
             &self.weth,
             self.unwrap_factor,
         )
-        .await
+        .await;
+
+        match score_calculator {
+            Some(score_calculator) => {
+                optimize_score(
```

Comment:
I would probably call it `compute_score()`. I agree that it kind of fits into the `PostProcessingPipeline` but just because the other steps are currently optimizations doesn't mean that this one also has to be called an optimization.

Reply:
renamed